### PR TITLE
feat(Scratch): add domains and refactor

### PR DESCRIPTION
**Pull request by Camuise**
* feat(Scratch): add domains
   - adds presence for individual wikis listed on [https://scratch-wiki.info](https://scratch-wiki.info)
   - previously only displayed generic "viewing wiki" presence

* feat(Scratch): merge domains
   - merges "scratch.mit.edu" and "scratch.mit.edu/" into one if statement
   - previously 2 if statements

* feat(Scratch): fix scratch project editor presence
   - checks only for "/editor" as it is the only path used on all editor pages
   - previously checked for "/projects/editor/" which is only accessed when entering the editor rather than loading/ using the editor causing to generic "scratch" presence

* refactor(Scratch): switch from window.location.hostname to window.location.href
   - affects "donationpay.org/scratchfoundation" and "scratched.gse.harvard.edu"
   - specifically consolidates "donationpay.org/scratchfoundation" from a window.location.hostname and window.location.pathname check into one window.location.href check

* style(Scratch): fix pretty-print
   - vs code underlined ~40% of the file as having pretty-print issues one day

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
https://imgur.com/a/NlhEev8
</details>

### CODE REVIEW

File: websites/S/Scratch/presence.ts [hunk 1]
```diff
@@ -239,23 +230,69 @@ presence.on("UpdateData", () => {
 			presenceData.state = "Scratch Foundation";
 		}
 	}
-	if (window.location.hostname.toLowerCase().includes("scratch-wiki.info")) {
-		presenceData.details = "Viewing page:";
-		presenceData.state = "Wiki";
+	if (window.location.href.toLowerCase().includes("scratch-wiki.info")) {
+		if (window.location.href.toLowerCase().includes("en.scratch-wiki.info")) {
```
Comment (Slowlife01) [inline websites/S/Scratch/presence.ts:234]: You can make this a switch case with `location.hostname`

File: websites/S/Scratch/presence.ts [hunk 2]
```diff
@@ -239,23 +230,69 @@ presence.on("UpdateData", () => {
...
+		window.location.href
+			.toLowerCase()
+			.includes("donationpay.org/scratchfoundation")
 	) {
 		presenceData.details = "Viewing page:";
 		presenceData.state = "Donate";
 	}
 	if (
-		window.location.hostname.toLowerCase().includes("scratched.gse.harvard.edu")
+		window.location.href.toLowerCase().includes("scratched.gse.harvard.edu")
```
Comment (Slowlife01) [inline websites/S/Scratch/presence.ts:295]: I don't see why you should use href here

File: websites/S/Scratch/presence.ts [hunk 3]
```diff
@@ -239,23 +230,69 @@ presence.on("UpdateData", () => {
 			presenceData.state = "Scratch Foundation";
 		}
 	}
-	if (window.location.hostname.toLowerCase().includes("scratch-wiki.info")) {
-		presenceData.details = "Viewing page:";
-		presenceData.state = "Wiki";
+	if (window.location.href.toLowerCase().includes("scratch-wiki.info")) {
```
Comment (Slowlife01) [inline websites/S/Scratch/presence.ts:233]: same as previous review

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/S/Scratch/dist/metadata.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
 	},
 	"url": "scratch.mit.edu",
 	"regExp": "([a-z0-9-]+[.])*(scratch|scratchfoundation|scratch-wiki|donationpay|scratched[.]gse[.]harvard|scratchjr)[.](mit[.]edu|org|info|edu)[/]",
-	"version": "1.2.12",
+	"version": "1.2.13",
```
Comment (EncryptedDev) [inline websites/S/Scratch/dist/metadata.json:14]: You are adding quite a bit of functionality, probably should bump minor instead
```suggestion
	"version": "1.3.0",
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/S/Scratch/presence.ts (hunk 2 above)
Comment (Marcus5408) [reply websites/S/Scratch/presence.ts:295]: i did some testing, it seems that href manages to detect url changes better

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**